src/styles/_calendar.scss

```diff
@@ -0,0 +1,45 @@
+body {
+  @include flex;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: $padding;
+  max-width: $calendar-width;
+  gap: $gap;
+
+  @each $day, $index in $weekdays {
+    &--start-day--#{$day} &__day:first-child {
+      margin-left: (($width + $gap) * $index);
+    }
+  }
+
+  @for $day-length from 28 through 31 {
+    &--month-length--#{$day-length} &__day:nth-child(n + #{$day-length + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    @include flex;
+    @include square($width, $height);
+    transition: all 0.5s;
```

Comment:
It is better to specify the exact properties instead of 'all'. It will be better for the performance
https://stackoverflow.com/questions/8947441/css3-transitions-is-transition-all-slower-than-transition-x
Also, it would be great to create a variable for transition duration